app/services/katello/pulp3/content_view_version/import.rb

```diff
@@ -50,6 +50,112 @@ def self.check!(content_view:, metadata:, path:, smart_proxy:)
                                path: path).check!
         end
 
+        def self.fetch_importable_products(organization:, metadata:)
```

Comment:
At minimum this method (and `fetch_importable_repositories`) should be renamed because it's doing quite a bit more than fetching. Maybe just `importable_products` is enough?

At most, you might think about these two methods in light of having to support RH products and repos next. Would supporting RH mean a rewrite? Maybe that can a refactor. The methods do a lot but there are not many branches thru the code which is good

Reply:
What do you think if I use something like -> https://refactoring.com/catalog/replaceFunctionWithCommand.html 

I can add a CustomRepoCreationHelper and a CustomProductCreationHelper under services ? I dont know if thats over engineering :)

Reply:
check out my latest commit and see if you like that direction